### PR TITLE
WIP UNTESTED: Refactor iqe test running

### PR DESCRIPTION
**Pull request by RonnyPfannschmidt**
* create a helper to run labeled iqe test runs
* create a helper to run collection gated testruns
* reshape transfering options to pytest args

### CODE REVIEW

**Comment by skarekrow** [comment]
Great PR! But I guess the initial question is "why?". We're ditching this entire method very soon, and it doesn't seem to make much sense to create possibly more instability in the short term in these environments. There's a fair amount of churn here, and maybe I'm wrong. Ultimately I'd leave this to @bsquizz as he architected most of these things.

Things seem fine other than some whitespace, but without testing this PR exhaustively I wouldn't feel super comfortable approving it.

**Comment by RonnyPfannschmidt** [comment]
@skarekrow i want to make the parallel testing optional for my outage tests, and the lineup for test invocation has been error prone to comepare/read 

what is it going to be replaced with, i'm happy to check the other details

File: vars/iqeUtils.groovy [hunk 1]
```diff
@@ -464,28 +430,33 @@ private def createTestStages(String appName, Map appConfig) {
...
 
-    stage("Install red-hat-internal-envs plugin") {
-        sh "iqe plugin install red-hat-internal-envs"
-    }
+    stage("Install dependencies") {
+        when {
+            expression { not appOptions['skipInstall'] }
+        }
+        stage("Install red-hat-internal-envs plugin") {
+            sh "iqe plugin install red-hat-internal-envs"
```
Comment (quarckster) [inline vars/iqeUtils.groovy:438]: I believe this stage can be removed. We include `red-hat-internal-envs` in `iqe-core` and update on every release.

**Review by quarckster — CHANGES_REQUESTED** [review]

File: vars/iqeUtils.groovy (hunk 1 above)
Comment (RonnyPfannschmidt) [reply vars/iqeUtils.groovy:436]: @quarckster im only leaving it in as thats how it currently is - once this works i can consider functionality changes beyond skipping all installs and

**Comment by skarekrow** [comment]
My understanding is this is being replaced by bonfire/clowder and friends @RonnyPfannschmidt, correct @bsquizz? If so my statement stands, I'd rather not see e2e-deploy get any less reliable before we swap :D

**Comment by bsquizz** [comment]
Sort of, I think the stage and prod test suites will still use this for awhile.

If we want to get this in, we definitely need to try out this branch of pipeline-lib on those test suite jobs. And maybe the advisor smoke test job as well.

**Comment by RonnyPfannschmidt** [comment]
closing this as the code moved into the images anyway

**Pull request closed without merging**